pkg/ddevapp/apptypes_test.go

```diff
@@ -44,3 +48,54 @@ func TestApptypeDetection(t *testing.T) {
 	}
 
 }
+
+// TestPostConfigAction tests that the post-config action is properly applied, but only if the
+// config is not included in the config.yaml.
+func TestPostConfigAction(t *testing.T) {
+	assert := asrt.New(t)
+
+	appTypes := map[string]string{
+		"drupal6":   "5.6",
+		"drupal7":   "7.1",
+		"drupal8":   ddevapp.DdevDefaultPHPVersion,
+		"wordpress": ddevapp.DdevDefaultPHPVersion,
+		"backdrop":  ddevapp.DdevDefaultPHPVersion,
+	}
+
+	for appType, expectedPHPVersion := range appTypes {
+		testDir := testcommon.CreateTmpDir("TestApptype")
+
+		// testcommon.Chdir()() and CleanupDir() checks their own errors (and exit)
+		defer testcommon.CleanupDir(testDir)
+		defer testcommon.Chdir(testDir)()
```

Comment:
This may be what you intend, but these deferred calls won't be executed until after the entire test is complete (when `TestPostConfigAction` returns) and not at the end of the block.

Reply:
Yes, that pattern is used all over the place, it's just what we did back in the day. The cleanup after test completion is fine, it's all just about cleaning up the various crap directories.